src/promptflow-tools/promptflow/tools/open_source_llm.py

```diff
@@ -354,58 +386,75 @@ def __call__(
+                if key not in conn_dict:
+                    accepted_keys = ",".join([key for key in self.REQUIRED_CONFIG_KEYS])
+                    raise OpenSourceLLMKeyValidationError(
+                        message=f"""Required key `{key}` not found in given custom connection.
 Required keys are: {accepted_keys}."""
-                )
-        for key in self.REQUIRED_SECRET_KEYS:
-            if key not in conn_dict:
-                accepted_keys = ",".join([key for key in self.REQUIRED_SECRET_KEYS])
-                raise OpenSourceLLMKeyValidationError(
-                    message=f"""Required secret key `{key}` not found in given custom connection.
+                    )
+            for key in self.REQUIRED_SECRET_KEYS:
+                if key not in conn_dict:
+                    accepted_keys = ",".join([key for key in self.REQUIRED_SECRET_KEYS])
+                    raise OpenSourceLLMKeyValidationError(
+                        message=f"""Required secret key `{key}` not found in given custom connection.
 Required keys are: {accepted_keys}."""
-                )
-        try:
-            self.model_family = ModelFamily[connection.configs['model_family']]
-        except KeyError:
-            accepted_models = ",".join([model.name for model in ModelFamily])
-            raise OpenSourceLLMKeyValidationError(
-                message=f"""Given model_family '{connection.configs['model_family']}' not recognized.
+                    )
+            try:
+                self.model_family = ModelFamily[connection.configs['model_family']]
+            except KeyError:
+                accepted_models = ",".join([model.name for model in ModelFamily])
+                raise OpenSourceLLMKeyValidationError(
+                    message=f"""Given model_family '{connection.configs['model_family']}' not recognized.
 Supported models are: {accepted_models}."""
-            )
-        self.connection = connection
+                )
+            self.endpoint_uri = connection.configs['endpoint_url']
+            self.endpoint_key = connection.secrets['endpoint_api_key']
 
     @tool
     @handle_oneline_endpoint_error()
     def call(
         self,
         prompt: PromptTemplate,
         api: API,
-        deployment_name: Optional[str] = None,
+        temperature: float = 1.0,
+        max_new_tokens: int = 200,
+        top_p: float = 1.0,
         model_kwargs: Optional[Dict] = {},
         **kwargs
     ) -> str:
         prompt = render_jinja_template(prompt, trim_blocks=True, keep_trailing_newline=True, **kwargs)
 
+        model_kwargs["top_p"] = top_p
+        model_kwargs["temperature"] = temperature
+        model_kwargs["max_new_tokens"] = max_new_tokens
+
         content_formatter = ContentFormatterFactory.get_content_formatter(
             model_family=self.model_family,
             api=api,
             chat_history=prompt
         )
 
         llm = AzureMLOnlineEndpoint(
```

Comment:
@chjinche , How does telemetry work here? We are interested in capturing few metrics like
- number of times this tool got called - success and failure rate
- number  of token used from this tool, model family choosen and chat/completion
- number  of prompt flow this tool is used in and sub/ws details
- number  of deployed prompt flow that are using this tool and sub/ws details.